static/js/giphy.js

```diff
@@ -107,11 +154,27 @@ export function initialize() {
         e.preventDefault();
         e.stopPropagation();
 
-        if ($("#giphy_grid_in_popover").length) {
-            $("#compose_box_giphy_grid").popover("hide");
+        if (active_popover_element && $.contains(active_popover_element.get()[0], e.target)) {
+            // Hide giphy popover if already active.
+            hide_giphy_popover();
             return;
         }
-        $("#compose_box_giphy_grid").popover("show");
+        popovers.hide_all();
+        const $elt = $(e.target);
+        // Store data-message-id value in global variable edit_message_id so that
+        // its value can be further used to correctly find the message textarea element.
+        edit_message_id = $elt.attr("data-message-id");
```

Comment:
Can you extend the comment here to mention that this is intentionally `undefined` if we're dealing with the compose box?  Otherwise, it looks like this should be a message-edit only block of code.

Reply:
Done.

Reply:
Cool, tweaked the language a bit before merging.